### PR TITLE
AN-6745: Migration between app versions freezes the app - Part 7

### PR DESCRIPTION
**Pull request by mejdoo**
## What's new in this PR?

Part 6 of #2679.

Migrated tables : Assets, Conversations, ConversationMembers, Messages, SyncJobs, SyncErrors
#### APK
[Download build #1576](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1576/artifact/build/artifact/wire-dev-PR2694-1576.apk)
[Download build #1580](http://10.10.124.11:8080/job/Pull%20Request%20Builder/1580/artifact/build/artifact/wire-dev-PR2694-1580.apk)

### CODE REVIEW

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala [hunk 1]
```diff
@@ -207,7 +207,6 @@ object UserData {
     val IntegrationId = opt(id[IntegrationId]('integration_id))(_.integrationId)
     val ExpiresAt = opt(remoteTimestamp('expires_at))(_.expiresAt)
     val Managed = opt(text[ManagedBy]('managed_by, _.toString, ManagedBy(_)))(_.managedBy)
-    val Fields = json[Seq[UserField]]('fields)(UserField.userFieldsDecoder, UserField.userFieldsEncoder)(_.fields)
```
Comment (gizemb) [inline wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala]: why did you erase that?

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt [hunk 2]
```diff
@@ -58,6 +64,231 @@ val USER_DATABASE_MIGRATION_126_TO_127 = object : Migration(126, 127) {
         createButtonsTable(database)
     }
 
+    private fun migrateUserTable(database: SupportSQLiteDatabase) {
+        val tempTableName = "UsersTemp"
+        val originalTableName = "Users"
+        val primaryKey = "_id"
+        val searchKey = "skey"
```
Comment (gizemb) [inline storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt]: why did we need `val`s for these 2?

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala (hunk 1 above)
Comment (mejdoo) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala]: no column in`Users` table with this name `fields`

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt (hunk 2 above)
Comment (BradleyWire) [reply storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt]: I was trying to create index statements (you'll see them in part 8) and must have pushed this branch for Mejdoo prematurely before implementing them, they're not redundant after this PR.

File: wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala (hunk 1 above)
Comment (gizemb) [reply wire-android-sync-engine/zmessaging/src/main/scala/com/waz/model/UserData.scala]: hmmm, ok.

File: storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt (hunk 2 above)
Comment (gizemb) [reply storage/src/main/kotlin/com/waz/zclient/storage/db/users/migration/UserDatabase126To127Migration.kt]: it's ok if they're being used in the next commits

**Pull request merged**